File: golang/src/p7/main.go
package main

import "fmt"
import "acommon"

func main() {
	primeIdx := 0
	current := 1

	for primeIdx < 10001 {
		current++
		if acommon.IsPrime(current) {
			primeIdx++
		}
	}

	fmt.Println(current)
}

File: cpp/src/p7/main.cc
#include <iostream>
#include "utils.h"

int main() {
  int primeIdx = 0;
  int current = 1;

  while (primeIdx < 10001) {
    ++current;
    if (isPrime(current)) {
      ++primeIdx;
    }
  }

  std::cout << current << "\n";
}
